modules/transactionpool/update.go

```diff
@@ -8,6 +8,124 @@ import (
 	"github.com/NebulousLabs/Sia/types"
 )
 
+// findSets takes a bunch of transactions (presumably from a block) and finds
+// all of the separate transaction sets within it. Set does not check for
+// conflicts.
+func findSets(ts []types.Transaction) [][]types.Transaction {
```

Comment:
just spitballing, but would this code be any simpler if we mandated that for any transaction set, transaction[i] can only reference transaction[j] if j < i? How easy would it be to detect transactions that didn't conform to this?
I'm wondering if placing a little restriction on txn ordering could reduce the computational load on the transaction pool.